### PR TITLE
refresh dev instruction to make a successful build

### PR DESCRIPTION
**Pull request by shawncao**
I have tried this change on a fresh new ubuntu 18 twice, it works fine. 
Could you follow dev.md from beginning until step 'build nebula", hope this time it works alright.

### CODE REVIEW

**Comment by shawncao** [comment]
Would like to see if this works for you, @samprasyork

File: dev.md [hunk 1]
```diff
@@ -1,36 +1,121 @@
...
+2. cd fmt
+3. cmake . && make -j$(nproc)
+4. sudo make install
+
+### Install BOOST
+1. sudo apt-get install -y libboost-dev
+
+### Install Facebook Folly
+1. git clone https://github.com/facebook/folly.git
+2. mkdir _build && cd _build
```
Comment (samprasyork) [inline dev.md]: do cd folloy before this step?

File: dev.md [hunk 2]
```diff
@@ -1,36 +1,121 @@
...
+
+### Install DOUBLE-CONVERSION
+1. git clone https://github.com/google/double-conversion.git
+2. cd double-conversion
+3. cmake -DBUILD_SHARED_LIBS=OFF . && make && sudo make install
+
+### Install GFLAGS, GLOG, GTEST
+1. sudo apt-get install -y libgflags-dev
+2. git clone https://github.com/google/glog.git
+3. cd glog && make . && sudo make install
```
[the remark this answers is not in the document]
Comment (shawncao) [reply dev.md:40]: I think you're right, I don't why I missed cmake step, will update.

File: dev.md (hunk 1 above)
Comment (shawncao) [reply dev.md]: yes, will update

File: dev.md [hunk 3]
```diff
@@ -1,36 +1,121 @@
...
+A local change to solve the the build break (in `~/nebula/build`):
+1. grpc/src/grpc/include/grpcpp/impl/codegen/byte_buffer.h
+   move method to public grpc_byte_buffer* c_buffer() { return buffer_; }
+2. flatbuffers/src/flatbuffers/include/flatbuffers/grpc.h
+   use ByteBuffer as parameter type of Deserialize
+     + static grpc::Status Deserialize(ByteBuffer *bb, flatbuffers::grpc::Message<T> *msg) {
+     + grpc_byte_buffer* buffer = nullptr;
+     + if (!bb || !(buffer = bb->c_buffer())) {
+         return ::grpc::Status(::grpc::StatusCode::INTERNAL, "No payload");
+       }
```
Comment (samprasyork) [inline dev.md:106]: How about move this section under “Install linker dependencies”, it is better to finish all the decencies first, and the modified the code.

Comment (samprasyork) [reply dev.md:106]: Actually, maybe under build nebula, after cmake ..

Comment (shawncao) [reply dev.md:101]: Added one more step before this to ensure both flatbuffers and grpc downloaded.

**Comment by shawncao** [comment]
All tests pass now with refresh on both Mac and Ubuntu 18. 

Screen Shot 2020-11-16 at 9 56 25 PM

Hey @samprasyork, can you approve this now?

**Comment by samprasyork** [comment]
Approved~

**Review by samprasyork — APPROVED** [review]

**Pull request merged**